# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/CreateElementTransaction.h
# Commit: d3816f82c0b9
# Full Hash: d3816f82c0b9d4ceca0bb2ad177694164ed0921f
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-04-03 21:40:10
# Regressor Bug: 1619914
# File Overlap Count: 1
# Description:
#   Bug 1619914 - part 2: Mark transaction class methods and their caller methods as `MOZ_CAN_RUN_SCRIPT r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D69303
# ==============================================================================

diff -r 876b1031a6ae -r d3816f82c0b9 editor/libeditor/CreateElementTransaction.h
--- a/editor/libeditor/CreateElementTransaction.h	Fri Apr 03 01:38:22 2020 +0000
+++ b/editor/libeditor/CreateElementTransaction.h	Fri Apr 03 01:39:36 2020 +0000
@@ -51,7 +51,7 @@
 
   NS_DECL_EDITTRANSACTIONBASE
 
-  NS_IMETHOD RedoTransaction() override;
+  MOZ_CAN_RUN_SCRIPT NS_IMETHOD RedoTransaction() override;
 
   dom::Element* GetNewElement() const { return mNewElement; }
 
@@ -61,7 +61,7 @@
   /**
    * InsertNewNode() inserts mNewNode before the child node at mPointToInsert.
    */
-  void InsertNewNode(ErrorResult& aError);
+  MOZ_CAN_RUN_SCRIPT void InsertNewNode(ErrorResult& aError);
 
   // The document into which the new node will be inserted.
   RefPtr<EditorBase> mEditorBase;